### PR TITLE
Ariel/Adding Default Text on Exercise Screen

### PR DESCRIPTION
**Pull request by ArielWangX**
If there is no exercise card in the database, the default text "Create Your First Exercise!" will be displayed on the ExerciseScreen.

### CODE REVIEW

File: app/src/main/java/com/arielwang/workoutlogger/database/daos/ExerciseCardDao.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ import com.arielwang.workoutlogger.database.model.ExerciseCard
 @Dao
 interface ExerciseCardDao {
     @Query("SELECT * FROM ExerciseCard")
-    suspend fun getAllCards(): List<ExerciseCard>
+    suspend fun getAllExerciseCards(): List<ExerciseCard>
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/database/daos/ExerciseCardDao.kt:9]: 👍

File: app/src/main/java/com/arielwang/workoutlogger/features/exercisecard/ui/ExerciseCardDestination.kt [hunk 2]
```diff
@@ -7,20 +7,20 @@ import com.arielwang.workoutlogger.core.ui.screen.screenViewModel
 import com.arielwang.workoutlogger.navigate.NavigationDestination
 import com.arielwang.workoutlogger.navigate.destination.composable
 
-object ExerciseDetailDestination: NavigationDestination {
+object ExerciseCardDestination: NavigationDestination {
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercisecard/ui/ExerciseCardDestination.kt:10]: 👍

File: app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt [hunk 3]
```diff
@@ -188,38 +197,47 @@ class TrackViewModel @Inject constructor(
...
 
-    // the value of TextField cannot be negative
-    private fun noNegativeNumber(text: String): String {
-        return if ((text.toInt() - 1) < 0) {
-            "0"
-        } else {
-            (text.toInt() - 1).toString()
-        }
+    // the minimum value of the TextField is zero
+    private fun coerceAtLeastZero(number: Int): Int {
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt:214]: Not sure if this is worth wrapping as a function. You can use coerceAtLeast directly instead

File: app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt [hunk 4]
```diff
@@ -188,38 +197,47 @@ class TrackViewModel @Inject constructor(
...
+        return number.coerceAtLeast(0)
     }
 
-    // Set maximum input length of TextField
-    private fun maximumTextFieldInput(
+    // Adding a condition to limit maximum input length in TextField
+    // if the length of user input is less or equal than maximum input length, input stay the same
+    // else, slice the user input length to maximum input length
+    //       and change the value of shouldShow to true.
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt:221]: nit: Formatting

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -3,8 +3,9 @@
 
     <string name="LandingScreen_title">Current time is: %1$s</string>
     <string name="LandingScreen_buttonContent">Next Screen</string>
-    <string name="HomeScreen_title">No Workout For Today</string>
+    <string name="HomeScreen_defaultText">No Workout For Today</string>
```
Comment (fredliao123) [inline app/src/main/res/values/strings.xml:6]: Does the homescreen only show today's workout? Maybe `No Workout Added` makes more sense?

File: app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt [hunk 6]
```diff
@@ -36,16 +37,23 @@ object TrackView {
...
 
-data class TextFieldProperty(
-    var text: String = "",
-    var maxChar: Boolean = false
+data class ShouldShowMaxCharAlert(
+    var shouldShow: Boolean = false,
+    var maxChar: Int = 0
+)
+
+data class ValidTextAndShouldShowMaxCharAlert(
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt:54]: Perhaps `ValidTextWithAlertState` is better?

**Review by fredliao123 — APPROVED** [review]
Great work, just a few small comments

File: app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt (hunk 3 above)
Comment (ArielWangX) [reply app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt:214]: Thank you.

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (ArielWangX) [reply app/src/main/res/values/strings.xml:6]: How about ```Create First Workout!``` ?

**Pull request merged**